### PR TITLE
Windows fixes post pr1718

### LINKED ISSUE #848: Windows: intermittent crash while attempting to adoptFileAsBucket
Unclear what is causing this.

the rename in BucketManagerImpl::adoptFileAsBucket fails with "Access denied".
When this happens, the function throws, and the system collapses.

It's unclear to me what is causing this intermittent problem.
A few potential candidates:
- file still open in a different thread (there used to be a race with the checksum thread that I fixed earlier, maybe there is something else)
- file still open in a different process (gzip?)
- anti-virus

If it's AV the only way to work around this might be to either sleep a bit before attempting the rename or make the "adoptFileAsBucket" retry-able.

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
# Description
This PR fixes a few problems with the windows build:
* there were some issues in the solution file causing certain files to not be in the right sub-folder
* the project didn't compile due to excessive number of symbols
* various warnings needed to be fixed
* a race condition was triggered when catching up ; Resolves #848

### CODE REVIEW

File: src/bucket/BucketManagerImpl.cpp [hunk 1]
```diff
@@ -184,7 +184,14 @@ BucketManagerImpl::adoptFileAsBucket(std::string const& filename,
...
             std::string err("Failed to rename bucket :");
             err += strerror(errno);
-            throw std::runtime_error(err);
+            // there is a race condition with external systems (windows)
+            // retry after sleeping for a second seems to work around the problem
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            if (rename(filename.c_str(), canonicalName.c_str()) != 0)
+            {
+                // if rename fails again, surface the original error
+                throw std::runtime_error(err);
```
Comment (jonjove) [inline src/bucket/BucketManagerImpl.cpp]: What message should the error contain? Should it be from the first failure, the second failure, or both? If it were to fail twice, I don't think it is necessarily true that both failures would be the same.

File: src/bucket/BucketManagerImpl.cpp [hunk 2]
```diff
@@ -184,7 +184,14 @@ BucketManagerImpl::adoptFileAsBucket(std::string const& filename,
         {
             std::string err("Failed to rename bucket :");
             err += strerror(errno);
-            throw std::runtime_error(err);
+            // there is a race condition with external systems (windows)
```
Comment (jonjove) [inline src/bucket/BucketManagerImpl.cpp]: If this is only known to occur on Windows, then I think this should be a conditional compile for Windows.

File: src/bucket/BucketManagerImpl.cpp (hunk 1 above)
Comment (MonsieurNicolas) [reply src/bucket/BucketManagerImpl.cpp]: I decided to use the original error as it's the reason we're failing in the first place

File: src/bucket/BucketManagerImpl.cpp (hunk 2 above)
Comment (MonsieurNicolas) [reply src/bucket/BucketManagerImpl.cpp]: I am not sure it's a Windows only problem: I run into it all the time on Windows, that's all I know. I'd rather keep the code the same on both platforms unless this causes problems

**Comment by jonjove** [comment]
r+ d483a7e

**Pull request merged**